renovation_core/utils/sms_setting.py

```diff
@@ -40,43 +46,90 @@ def send_sms(receiver_list, msg, sender_name='', success_msg=True):
       'message'		: frappe.safe_decode(msg).encode('utf-8'),
       'success_msg'	: success_msg
   }
-
-  if frappe.db.get_value('SMS Settings', None, 'sms_gateway_url'):
-    return send_via_gateway(arg)
-  else:
-    msgprint(_("Please Update SMS Settings"))
-    return "Fail"
+  if not provider:
+    provider = get_default_sms_providers()
+  elif isinstance(provider, string_types):
+    provider = [{"provider": provider, "country": x.country, "code": x.code.lower()} for x in (
+        frappe.get_cached_doc("SMS Provider", provider).get("countries") or [])]
+    provider.append({"provider": provider, "country": "all", "code": "all"})
+  return send_via_gateway(arg, provider=provider)
 
 
-def send_via_gateway(arg):
-  ss = frappe.get_doc('SMS Settings', 'SMS Settings')
-  headers = get_headers(ss)
+def get_default_sms_providers():
+  system_settings = frappe.get_single("System Settings")
+  return system_settings.sms_providers
 
-  args = {ss.message_parameter: re.sub(
-      r'\s+', ' ', safe_decode(arg.get('message')))}
-  for d in ss.get("parameters"):
-    if not d.header:
-      args[d.parameter] = d.value
 
+def send_via_gateway(arg, provider):
```

Comment:
@mainul94 I think better to have it plural since it can be multiple.
```suggestion
def send_via_gateway(arg, providers):
```